openpype/hosts/fusion/hooks/pre_fusion_setup.py

```diff
@@ -48,14 +138,26 @@ def execute(self):
         # Fusion 16 and 17 use FUSION16_PYTHON36_HOME instead of
         # FUSION_PYTHON3_HOME and will only work with a Python 3.6 version
         # TODO: Detect Fusion version to only set for specific Fusion build
-        self.launch_context.env["FUSION16_PYTHON36_HOME"] = py3_dir
+        self.launch_context.env[f"FUSION{self.VERSION}_PYTHON36_HOME"] = py3_dir  # noqa
 
-        # Add our Fusion Master Prefs which is the only way to customize
-        # Fusion to define where it can read custom scripts and tools from
+        # Add custom Fusion Master Prefs and the temporary
+        # profile directory variables to customize Fusion
+        # to define where it can read custom scripts and tools from
         self.log.info(f"Setting OPENPYPE_FUSION: {FUSION_HOST_DIR}")
         self.launch_context.env["OPENPYPE_FUSION"] = FUSION_HOST_DIR
-
-        pref_var = "FUSION16_MasterPrefs"   # used by Fusion 16, 17 and 18
-        prefs = os.path.join(FUSION_HOST_DIR, "deploy", "fusion_shared.prefs")
-        self.log.info(f"Setting {pref_var}: {prefs}")
-        self.launch_context.env[pref_var] = prefs
+        (
+            copy_status,
+            fusion_profile_dir,
+            force_sync,
+        ) = self.get_copy_fusion_prefs_settings()
+        if copy_status and fusion_profile_dir is not None:
+            prefs_source = self.get_profile_source()
+            self.copy_existing_prefs(prefs_source, fusion_profile_dir, force_sync)  # noqa
+        else:
+            fusion_profile_dir_variable = f"FUSION{self.VERSION}_PROFILE_DIR"
+            self.log.info(f"Setting {fusion_profile_dir_variable}: {fusion_profile_dir}")  # noqa
+            self.launch_context.env[fusion_profile_dir_variable] = str(fusion_profile_dir)  # noqa
+        master_prefs_variable = f"FUSION{self.VERSION}_MasterPrefs"
```

Comment:
```suggestion

        master_prefs_variable = f"FUSION{self.VERSION}_MasterPrefs"
```

I personally really prefer a little breathing space around some lines of code. For example here, to show that the master prefs logic here isn't one-to-one related directly the copying of the fusion profile dir.

Actually looking at the code involved with the copying of the Fusion profile dir I'm actually thinking of moving that full logic into a PreLaunchHook of its own.

So we have a dedicated one for setting the PYTHONHOME and a dedicated one for the potential copying of user prefs. So separating to a dedicated `FusionCopyPrefsPrelaunch` - anyone else feel like that separating those responsibilities would make for clearer code?

There's so much code here that looks intertwined and related to each other but in the end it doesn't rely on each other at all.

Reply:
Do you mean of adding another prelaunch hook module just for the profile? I was thinkng about that, but could not get it to work. How should I trigger the new hook module? 
![image](https://user-images.githubusercontent.com/11698866/223551065-ed8479fe-4fbe-4f84-925e-179e34224196.png)

Reply:
Yes. Just copy paste this file in the same folder, change the class name, etc. and separate the logic between the two. That should roughly do it.